Azure/azure-cli
# [App Service] `az functionapp deployment source config-zip`: Fix returning success when the deployment did not succeed

**Description**
Fixes an issue with `az functionapp deployment source config-zip` when Kudu returns a 5xx response code instead of the expected 202.

Before this change:
* If Kudu responds with a 5xx response code to the /api/zipdeploy endpoint, Azure CLI continues anyway, and requests the latest deployment log
* The latest deployment log contains whatever deployment was done _before_ this one, and was likely successful
* So the command overall succeeds (exit code 0)

After this change:
* If Kudu responds with a 5xx response code to the /api/zipdeploy endpoint, Azure CLI immediately ends with an error because the deployment failed

**Testing Guide**
1. Run a command like `az functionapp deployment source config-zip -g MyResourceGroup -n MyFunctionApp --src-file source.zip`
   * Also applies to `az webapp deployment source config-zip` since the two commands share this code.
2. Note in the stderr output the line "WARNING: Deployment endpoint responded with status code 999"
3. If the status code is anything other than 202, the command should have printed an error to stderr and ended with a non-zero exit code.

---

This checklist is used to make sure that common guidelines for a pull request are followed.
